### PR TITLE
Fix notInRangeMessage usage for a few Range validations

### LINKED ISSUE #13883: [Admin][Cart Promotions] Coupons based - generate coupons - error message not showing the correct limit.
**Sylius version affected**: 1.11.3

**Description**  
Generate cart promo actions with coupons based
Generate coupons enter `Code length` with number 1024.
I see error message:

Coupon code must not be longer than {{ limit }} characters. 

The limit is not specified. Not sure what the limit is.

**Steps to reproduce**  
Generate cart promo actions with coupons based
Generate coupons enter `Code length` with number 1024.

**Possible Solution**  
Correct the translation limit.

### PR DESCRIPTION
**Pull request by diimpp**
| Q               | A                                                            |
|-----------------|--------------------------------------------------------------|
| Branch?         | 1.11                  |
| Bug fix?        | yes                                                       |
| New feature?    | no                                                       |
| BC breaks?      | no                                                       |
| Deprecations?   | yes  |
| Related tickets | fixes #13883                       |
| License         | MIT                                                          |

**Issue**
Symfony validation constraint [`Range`](https://symfony.com/doc/4.4/reference/constraints/Range.html#notinrangemessage) instroduced new behavior since 4.4 release -- if both `min` and `max` options are defined, then `notInRangeMessage` will be shown and `minMessage`/`maxMessage` will be disregarded.

<details><summary>Example</summary>

If you will try to generate coupons for promotion with code length larger than 40, then you will see symfony standard message `notInRangeMessage: This value should be between {{ min }} and {{ max }}.` instead of sylius `maxMessage: Coupon code must not be longer than {{ limit }} characters.`
image

</details>

#### Case 1

https://github.com/Sylius/Sylius/blob/9506c80f2080b34b89f51b54e7e8f3490c8b7d36/src/Sylius/Bundle/PromotionBundle/Resources/config/validation/PromotionCouponGeneratorInstruction.xml#L33-L37

- [x] Remove minMessage, maxMessage and their translations
- [x] Introduce notInRangeMessage and new translation
- [x] Cover case with behat scenario 

#### Case 2
https://github.com/Sylius/Sylius/blob/9506c80f2080b34b89f51b54e7e8f3490c8b7d36/src/Sylius/Bundle/PromotionBundle/Form/Type/Action/UnitPercentageDiscountConfigurationType.php#L35-L41

- [x] Remove minMessage, maxMessage usage; Their translations were already removed.
- [x] Introduce notInRangeMessage; Translation already exists as it's shared between order/item percentage discount
- [x] Cover case with behat scenario

#### Case 3
https://github.com/Sylius/Sylius/blob/9506c80f2080b34b89f51b54e7e8f3490c8b7d36/src/Sylius/Bundle/ApiBundle/Resources/config/validation/AddProductReview.xml#L23-L28

- [x] Remove minMessage, maxMessage
- [x] Rename translation key `range` to `not_in_range`
- [x] Move to separate https://github.com/Sylius/Sylius/pull/14584 due deprecation.

**Questions**
1. Does removal of `sylius.promotion_coupon_generator_instruction.code_length.min` and `...max` messages constitute a deprecation entry? Because those translation strings were inactive (effectively deprecated by symfony upgrade) since project was upgraded to symfony 4.4 and even if somebody has custom translation over there, it has no effect whatsoever. 
2. Please advise on adding/removal of translation strings, do I need to do anything else? I do recall some translation service was in use.

### CODE REVIEW

**Review by Zales0123 — COMMENTED** [review]
Nice change, @diimpp! I would leave these min/max messages, as someone can use them in other places in their application. I would also add an UPGRADE file note about it and it's good to be merged 🏅 🎉

**Comment by diimpp** [comment]
@Zales0123 Hi, can we mention removal of translation min,max messages as BC break instead? Or is it a part of hard BC policy?

My point is why leave dead <s>code</s>message, if it's unused by the app since `1.6.0`. (symfony/*: ^3.4|^4.3)
While it's possible, that somebody made use of `Coupon code must be at least {{ limit }} characters long.` in their custom promotion validation, it doesn't look likely.

But I will update PR to keep them, if it's required.

P.S. In related https://github.com/Sylius/Sylius/pull/14584 I've changed `sylius.review.rating.range` to `sylius.review.rating.not_in_range` -- Do I keep old trans key as well?

**Comment by jakubtobiasz** [comment]
Hi @diimpp!
Thanks for your contribution. We decided we should leave the message keys. We don't use them, maybe you aren't either. But it doesn't change in some projects they might be used across the system, and we shouldn't break these places. Thanks in advance!

File: features/promotion/managing_promotions/promotion_validation.feature [hunk 1]
```diff
@@ -63,21 +63,31 @@ Feature: Promotion validation
         And promotion with name "Christmas sale" should not be added
 
     @ui @javascript
-    Scenario: Trying to add a new promotion with a wrong percentage discount
+    Scenario: Trying to add a new promotion without specifying a item percentage discount
```
Comment (GSadee) [inline features/promotion/managing_promotions/promotion_validation.feature]:
```suggestion
    Scenario: Trying to add a new promotion without specifying an item percentage discount
```
and it seems that this scenario is not related to the Range constraint 🤔

File: src/Sylius/Behat/Context/Ui/Admin/ManagingPromotionCouponsContext.php [hunk 2]
```diff
@@ -359,6 +359,14 @@ public function iShouldBeNotifiedThatCodeLengthIsRequired()
         Assert::true($this->generatePage->checkCodeLengthValidation('Please enter coupon code length.'));
     }
 
+    /**
+     * @Then I should be notified that generate code length is out of range
+     */
+    public function iShouldBeNotifiedThatCodeLengthIsOutOfRange()
```
Comment (GSadee) [inline src/Sylius/Behat/Context/Ui/Admin/ManagingPromotionCouponsContext.php]:
```suggestion
    public function iShouldBeNotifiedThatCodeLengthIsOutOfRange(): void
```

File: src/Sylius/Bundle/PromotionBundle/Resources/translations/validators.ar.yml [hunk 3]
```diff
@@ -31,8 +31,6 @@ sylius:
             min: يجب أن يكون مقدار كوبونات لتوليد على الأقل {{ limit }}.
             not_blank: الرجاء إدخال قيمة القسائم لتوليد.
         code_length:
-            min: البريد الإلكتروني يجب أن يكون على الأقل {{ limit }} حرفاً.
-            max: البريد الإلكتروني يجب أن لا تكون أطول من أحرف {{ limit }}.
```
Comment (GSadee) [inline src/Sylius/Bundle/PromotionBundle/Resources/translations/validators.ar.yml]: Removing these messages should be reverted

File: src/Sylius/Behat/Context/Ui/Admin/ManagingPromotionsContext.php [hunk 4]
```diff
@@ -513,8 +513,7 @@ public function iShouldBeNotifiedThatThisValueShouldNotBeBlank()
     }
 
     /**
-     * @Then I should be notified that a percentage discount value must be at least 0%
-     * @Then I should be notified that the maximum value of a percentage discount is 100%
```
Comment (GSadee) [inline src/Sylius/Behat/Context/Ui/Admin/ManagingPromotionsContext.php]: I would be for leaving these two steps as they are still valid and someone may have used them in the end application

File: src/Sylius/Bundle/PromotionBundle/Resources/translations/validators.th.yml [hunk 5]
```diff
@@ -31,8 +31,8 @@ sylius:
             min: จำนวนการสร้างคูปองต้องระบุอย่างน้อย {{ limit }}
             not_blank: กรุณาระบุยอดรวมในการสร้างคูปอง
         code_length:
+            not_blank: กรุณาระบุความยาวของรหัสคูปอง
             min: รหัสคูปองต้องมีความยาวไม่น้อยกว่า {{ limit }} อักษร
             max: รหัสคูปองต้องมีความยาวไม่เกิน {{ limit }} ตัวอักษร
-            not_blank: กรุณาระบุความยาวของรหัสคูปอง
         usage_limit:
```
Comment (NoResponseMate) [inline src/Sylius/Bundle/PromotionBundle/Resources/translations/validators.th.yml]: Should be reverted

**Review by NoResponseMate — APPROVED** [review]

File: src/Sylius/Bundle/PromotionBundle/Resources/translations/validators.th.yml [hunk 6]
```diff
@@ -31,8 +31,8 @@ sylius:
             min: จำนวนการสร้างคูปองต้องระบุอย่างน้อย {{ limit }}
             not_blank: กรุณาระบุยอดรวมในการสร้างคูปอง
         code_length:
+            not_blank: กรุณาระบุความยาวของรหัสคูปอง
```
Comment (TheMilek) [inline src/Sylius/Bundle/PromotionBundle/Resources/translations/validators.th.yml]: Why changed `not_blank` if it's in correct alphabetical order?

File: src/Sylius/Bundle/PromotionBundle/Resources/translations/validators.uk.yml [hunk 7]
```diff
@@ -34,9 +34,9 @@ sylius:
             min: Кількість створених купонів має бути як мінімум {{ limit }}.
             not_blank: Будь ласка, введіть кількість купонів для створення.
         code_length:
+            not_blank: Будь ласка, введіть довжину коду купона.
```
Comment (TheMilek) [inline src/Sylius/Bundle/PromotionBundle/Resources/translations/validators.uk.yml]: as above

File: src/Sylius/Bundle/PromotionBundle/Resources/translations/validators.zh_CN.yml [hunk 8]
```diff
@@ -31,8 +31,8 @@ sylius:
             min: 待生成的优惠券数量必须至少为 {{ limit }}。
             not_blank: 请输入想要生成的优惠券的数量。
         code_length:
+            not_blank: 优惠券代码不能为空。
```
Comment (TheMilek) [inline src/Sylius/Bundle/PromotionBundle/Resources/translations/validators.zh_CN.yml]: as above

**Review by TheMilek — APPROVED** [review]

**Review by jakubtobiasz — APPROVED** [review]

**Comment by TheMilek** [comment]
Thank you, Dmitri! :tada:
